src/extension/background-script/actions/webln/keysendOrPrompt.ts

```diff
@@ -0,0 +1,24 @@
+import { Message } from "../../../../types";
+import {
+  checkAllowance,
+  sendPaymentWithAllowance,
+  payWithPrompt,
+} from "./sendPaymentOrPrompt";
+
+const keysendOrPrompt = async (message: Message) => {
+  const destination = message.args.destination;
+  const amount = message.args.amount;
+  if (typeof destination !== "string" || typeof amount !== "string") {
+    return {
+      error: "Destination or amount missing.",
+    };
+  }
+
+  if (await checkAllowance(message.origin.host, parseInt(amount))) {
+    return sendPaymentWithAllowance(message, true);
```

Comment:
I am not a fan of magic boolean flags. When I read this code here it is unclear what this `true` means. 
looking into `sendPaymentWithAllowance` this function basically now does two things based on the boolean flag. 
I would split this up and use here a new `keySendWithAllowance` method. it is more descriptive and we don't need to add some `if` in those methods sendPayment methods